File: app/src/main/java/com/example/myapplication/SignUp.java
package com.example.myapplication;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class SignUp extends AppCompatActivity {
    Button btn;
    EditText name,uname,email,pwd;
    String url="https://museumapp69.000webhostapp.com/";
    TextView btnSignup;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_signup);


        name = (EditText) findViewById(R.id.name);
        uname = (EditText) findViewById(R.id.username);
        pwd = (EditText) findViewById(R.id.password);
        email = (EditText) findViewById(R.id.email);
        btnSignup = findViewById(R.id.loginText);


                btn =(Button) findViewById(R.id.regbutton);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if(TextUtils.isEmpty(name.getText().toString()) || TextUtils.isEmpty(uname.getText().toString()) || TextUtils.isEmpty(pwd.getText().toString()) || TextUtils.isEmpty(email.getText().toString())){
                    Toast.makeText(SignUp.this,"All fields Required", Toast.LENGTH_LONG).show();
                }else{
                    //proceed to reg
                    process();
                }
            }
        });

        btnSignup.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(SignUp.this, Login.class));
            }
        });
    }

    public void process(){
        Call<LoginResponse> loginResponseCall = ApiClient.getUserService().addData(name.getText().toString(), email.getText().toString(), uname.getText().toString(), pwd.getText().toString());
        loginResponseCall.enqueue(new Callback<LoginResponse>() {
            @Override
            public void onResponse(Call<LoginResponse> call, Response<LoginResponse> response) {

                if(response.isSuccessful()){
                    Toast.makeText(SignUp.this,"Registration Successful", Toast.LENGTH_LONG).show();

                    new Handler().postDelayed(new Runnable() {
                        @Override
                        public void run() {

                            startActivity(new Intent(SignUp.this, Login.class));
                        }
                    },700);

                }else{
                    Toast.makeText(SignUp.this,"Reg Failed", Toast.LENGTH_LONG).show();

                }

            }

            @Override
            public void onFailure(Call<LoginResponse> call, Throwable t) {
                Toast.makeText(SignUp.this,"Throwable "+t.getLocalizedMessage(), Toast.LENGTH_LONG).show();

            }
        });
    }
}

File: app/src/main/java/com/example/mobiliojiaplikacija/Signup.kt
package com.example.mobiliojiaplikacija

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.View
import android.widget.Button
import android.widget.ProgressBar
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import com.google.android.material.textfield.TextInputEditText
import com.vishnusivadas.advanced_httpurlconnection.PutData
import java.util.jar.Manifest


class Signup : AppCompatActivity() {




    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_signup)
        val textInputEditTextname:TextInputEditText = findViewById(R.id.name)
        val textInputEditTextusername:TextInputEditText = findViewById(R.id.username)
        val textInputEditTextpassword:TextInputEditText = findViewById(R.id.password)
        val textInputEditTextemail:TextInputEditText = findViewById(R.id.email)
        val buttonSignup:Button = findViewById(R.id.buttonSignUp)
        val textviewLogin:TextView = findViewById(R.id.loginText)
        val progressBar:ProgressBar = findViewById(R.id.progress)

        Toast.makeText(getApplicationContext(), "OOF", Toast.LENGTH_SHORT).show()

        buttonSignup.setOnClickListener(object: View.OnClickListener{
            override fun onClick(p0: View?) {

                var name:String = textInputEditTextname.getText().toString()
                var username:String = textInputEditTextusername.getText().toString()
                var password:String = textInputEditTextpassword.getText().toString()
                var email:String = textInputEditTextemail.getText().toString()

                if(!name.equals("") && !username.equals("") && !password.equals("") && !email.equals("")) {
                    progressBar.setVisibility(View.VISIBLE)
                    val handler = Handler(Looper.getMainLooper())
                    handler.post(Runnable {
                        //Starting Write and Read data with URL
                        //Creating array for parameters
                        val field = arrayOfNulls<String>(4)
                        field[0] = "name"
                        field[1] = "username"
                        field[2] = "password"
                        field[3] = "email"
                        //Creating array for data
                        val data = arrayOfNulls<String>(4)
                        data[0] = name
                        data[1] = username
                        data[2] = password
                        data[3] = email
                        val putData = PutData(
                            "https://museumapp69.000webhostapp.com/signup.php",
                            "POST",
                            field,
                            data
                        )
                        if (putData.startPut()) {
                            if (putData.onComplete()) {
                                progressBar.setVisibility(View.GONE)
                                val result = putData.result
                                if(result.equals("Sign Up Success")){

                                    val intent:Intent = Intent(getApplicationContext(), Login::class.java)
                                    startActivity(intent)
                                    finish()

                                }
                                else{

                                }
                            }
                        }
                        //End Write and Read data with URL
                    })
                }


            }
        })

    }
}